worlds/banjo_tooie/Regions.py

```diff
@@ -0,0 +1,2243 @@
+        region_map[regionName.IOHWH].append(locationName.W1)
+        region_map[regionName.IOHWH].append(locationName.W2)
+        region_map[regionName.IOHWH].append(locationName.W3)
+        region_map[regionName.IOHWH].append(locationName.W4)
+        region_map[regionName.IOHWH].append(locationName.W5)
+        region_map[regionName.IOHWH].append(locationName.W6)
+        region_map[regionName.IOHWH].append(locationName.W7)
+        region_map[regionName.IOHWH].append(locationName.W8)
+        region_map[regionName.IOHWH].append(locationName.W9)
+
+    if self.options.nestsanity:
+        nest_map = copy.deepcopy(NEST_REGIONS)
+        for region, locations in nest_map.items():
+            for location in locations:
+                region_map[region].append(location)
+
+    if self.options.randomize_signposts:
+        signpost_map = copy.deepcopy(SIGNPOST_REGIONS)
+        for region, locations in signpost_map.items():
+            for location in locations:
+                region_map[region].append(location)
+
+    if self.options.randomize_silos:
+        silo_map = copy.deepcopy(SILO_REGIONS)
+        for region, locations in silo_map.items():
+            for location in locations:
+                region_map[region].append(location)
+
+    if self.options.randomize_warp_pads:
+        warp_map = copy.deepcopy(WARP_PAD_REGIONS)
+        for region, locations in warp_map.items():
+            for location in locations:
+                region_map[region].append(location)
+
+    if self.options.randomize_tickets:
+        ticket_map = copy.deepcopy(BTTICKETS_REGIONS)
+        for region, locations in ticket_map.items():
+            for location in locations:
+                region_map[region].append(location)
+
+    if self.options.randomize_green_relics:
+        relic_map = copy.deepcopy(GREEN_RELIC_REGIONS)
+        for region, locations in relic_map.items():
+            for location in locations:
+                region_map[region].append(location)
+
+    if self.options.randomize_beans:
+        beans_map = copy.deepcopy(BEANS_REGIONS)
+        for region, locations in beans_map.items():
+            for location in locations:
+                region_map[region].append(location)
+
+    self.multiworld.regions.extend(create_region(
+        self.multiworld,
+        self.player,
+        active_locations,
+        region,
+        locations
+    ) for region, locations in region_map.items())
+    if self.options.victory_condition in (
+                VictoryCondition.option_hag1,
+                VictoryCondition.option_wonderwing_challenge,
+                VictoryCondition.option_boss_hunt_and_hag1
+            ):
+        self.multiworld.get_location(
+            locationName.HAG1, player
+        ).place_locked_item(
+            self.create_event_item(itemName.VICTORY)
+        )
+
+
+def create_region(multiworld, player: int, active_locations, name: str, locations=None):
+    ret = Region(name, player, multiworld)
```

Comment:
```suggestion
def create_region(world, active_locations, name: str, locations=None):
    ret = Region(name, world.player, world.multiworld)
```
Seems like it would probably make more sense to pass world here and then use that instead of `multiworld.worlds[player]`. Also it seems like the if chain could be replaced by an `options.victory_condition in` similar to above.